qiskit/circuit/quantumcircuit.py

```diff
@@ -4120,6 +4122,664 @@ def pauli(
+                    "The target qubit is a QuantumRegister containing more than one qubit."
+                )
+        # Check if q_controls has type "list"
+        if not isinstance(angle_list, list):
+            raise QiskitError("The angles must be provided as a list.")
+        num_contr = math.log2(len(angle_list))
+        if num_contr < 0 or not num_contr.is_integer():
+            raise QiskitError(
+                "The number of controlled rotation gates is not a non-negative power of 2."
+            )
+        # Check if number of control qubits does correspond to the number of rotations
+        if num_contr != len(q_controls):
+            raise QiskitError(
+                "Number of controlled rotations does not correspond to the number of control-qubits."
+            )
+        return self.append(UCRZGate(angle_list), [q_target] + q_controls, [])
+
+    @deprecate_func(
+        since="0.45.0", additional_msg="Instead, use the QuantumCircuit.unitary method."
+    )
+    def squ(
+        self,
+        unitary_matrix,
+        qubit,
+        mode="ZYZ",
+        up_to_diagonal=False,
+    ):
+        """Decompose an arbitrary 2*2 unitary into three rotation gates.
+
+        Note that the decomposition is up to a global phase shift.
+        (This is a well known decomposition which can be found for example in Nielsen and Chuang's book
+        "Quantum computation and quantum information".)
+
+        Args:
+            unitary_matrix (ndarray): 2*2 unitary (given as a (complex) ndarray).
+            qubit (QuantumRegister or Qubit): The qubit which the gate is acting on.
+            mode (string): determines the used decomposition by providing the rotation axes.
+                The allowed modes are: "ZYZ" (default)
+            up_to_diagonal (bool):  if set to True, the single-qubit unitary is decomposed up to
+                a diagonal matrix, i.e. a unitary u' is implemented such that there exists a 2*2
+                diagonal gate d with u = d.dot(u')
+
+        Returns:
+            InstructionSet: The single-qubit unitary instruction attached to the circuit.
+
+        Raises:
+            QiskitError: if the format is wrong; if the array u is not unitary
+        """
+        # pylint: disable=cyclic-import
+        from qiskit.extensions.quantum_initializer.squ import SingleQubitUnitary
+
+        if isinstance(qubit, QuantumRegister):
+            qubit = qubit[:]
+            if len(qubit) == 1:
+                qubit = qubit[0]
+            else:
+                raise QiskitError(
+                    "The target qubit is a QuantumRegister containing more than one qubit."
+                )
+        # Check if there is one target qubit provided
+        if not isinstance(qubit, Qubit):
+            raise QiskitError("The target qubit is not a single qubit from a QuantumRegister.")
+        return self.append(SingleQubitUnitary(unitary_matrix, mode, up_to_diagonal), [qubit], [])
```

Comment:
If I am not mistaken, this line will raise a second deprecation warning as per `qiskit/extensions/quantum_initializer/squ.py L45`, I see that you catch the deprecation warning for `snapshot`. Shouldn't you have a similar filter here?

Reply:
well, my experience with opflow deprecation warnings tells me that they are not as innocent as they seem (they can break unit tests from other libraries that depend on qiskit), so better safe than sorry :)

Reply:
Fixed in 80ace24